### PR TITLE
add option output path

### PR DESCRIPTION
**Pull request by b1nj**
Hello,
Can you check this pull request for change the default output path.

### CODE REVIEW

**Comment by b1nj** [comment]
I don't understand the line : 
https://github.com/Vishal0203/file-include-webpack-plugin/blob/9285ba7f2dfe3a2304c3a18e076475265b0ecf42/src/index.js#L49
Why save the file when it's done after ?

delete the ligne, the function saveFile and change line 55 by `size: () => content.length`

**Comment by Vishal0203** [comment]
might as well change the entire function to

```
process(compilation, callback) {
    const { context } = this.compiler.options
    this.context = path.join(context, this.source)
    const files = utils.getRequiredFiles(this.context, '')

    utils.logger.info(`Working on ${files.length} .html files`)

    files.forEach(file => {
      const sourcePath = path.join(this.context, file)
      const content = this.processFile(compilation, sourcePath)

      compilation.assets[file] = {
        source: () => content,
        size: () => content.length
      }
    })

    callback()
  }
```

I still need to test your changes. I feel it will break a few things. let me verify that first. 
Meanwhile, you can raise a new PR for the above mentioned change.

File: src/index.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ class FileIncludeWebpackPlugin {
     this.source = config.source // source from the context
     this.replace = config.replace
     this.context = null
+    this.outputPath = config.outputPath
```
Comment (Vishal0203) [inline src/index.js]: change `outputPath` to `destination`, to match the terminology

File: src/index.js [hunk 2]
```diff
@@ -37,14 +38,15 @@ class FileIncludeWebpackPlugin {
 
   process(compilation, callback) {
     const { context, output } = this.compiler.options
+    const outputPath = this.outputPath !== undefined ? this.outputPath : output.path;
```
Comment (Vishal0203) [inline src/index.js]: remove this line

File: src/index.js [hunk 3]
```diff
@@ -37,14 +38,15 @@ class FileIncludeWebpackPlugin {
...
     this.context = path.join(context, this.source)
     const files = utils.getRequiredFiles(this.context, '')
 
     utils.logger.info(`Working on ${files.length} .html files`)
 
     files.forEach(file => {
       const sourcePath = path.join(this.context, file)
-      const destinationPath = path.join(output.path, file)
+      const destinationPath = path.join(outputPath, file)
```
Comment (Vishal0203) [inline src/index.js]: change this to 
`const destinationPath = this.destination ? path.join(this.destination, file) : file`

File: src/index.js [hunk 4]
```diff
@@ -50,7 +50,7 @@ class FileIncludeWebpackPlugin {
       const content = this.processFile(compilation, sourcePath)
       const size = utils.saveFile(destinationPath, content)
 
-      compilation.assets[file] = {
+      compilation.assets[destinationPath] = {
```
Comment (Vishal0203) [inline src/index.js]: this will work fine you fix the review comments on the previous commit

File: README.md [hunk 5]
```diff
@@ -37,6 +37,7 @@ module.exports = {
     new FileIncludeWebpackPlugin(
       {
         source: './src/templates', // relative path to your templates
+        outputPath: '../html',
```
Comment (Vishal0203) [inline README.md]: Add this as another section in read me under the heading `How to change the output destination?`.
Do not change the default configuration readme.
change `outputPath` to `destination` and add the following statement.

> `destination` is an optional configuration which defaults to `output.path` in webpack config.

**Comment by Vishal0203** [comment]
> I don't understand the line :
> [file-include-webpack-plugin/src/index.js](https://github.com/Vishal0203/file-include-webpack-plugin/blob/9285ba7f2dfe3a2304c3a18e076475265b0ecf42/src/index.js#L49)
> 
> Line 49 in [9285ba7](/Vishal0203/file-include-webpack-plugin/commit/9285ba7f2dfe3a2304c3a18e076475265b0ecf42)
> 
>  const size = utils.saveFile(destinationPath, content) 
> 
> Why save the file when it's done after ?
> delete the ligne, the function saveFile and change line 55 by `size: () => content.length`

Agreed. 
you can remove it as part of this PR. 
Also, make sure you delete the helper functions from `utils.js`

File: src/index.js [hunk 6]
```diff
@@ -38,21 +38,20 @@ class FileIncludeWebpackPlugin {
 
   process(compilation, callback) {
     const { context, output } = this.compiler.options
-    const outputPath = this.outputPath !== undefined ? this.outputPath : output.path;
+    const outputPath = this.destination !== undefined ? this.destination : output.path;
```
Comment (Vishal0203) [inline src/index.js]: @b1nj 
this line is not needed since L49 does the job.

File: src/utils.js [hunk 7]
```diff
@@ -47,15 +47,6 @@ function mkdirSync(path) {
   }
 }
 
-function saveFile(path, content) {
```
Comment (Vishal0203) [inline src/utils.js]: also, delete `mkdirSync` function as it is not usable anymore.

**Comment by Vishal0203** [comment]
@b1nj once resolved the final 2 comments, please [squash](https://makandracards.com/makandra/527-squash-several-git-commits-into-a-single-commit) your commits into 1. We can release a patch post that. 

Thank you again!

**Comment by b1nj** [comment]
Sorry I do not understand how to do squash. I work in my master branch, so the article in makandracards it's not for me.

**Pull request merged**